### ISSUE TITLE
Refactor the grafana dashboards to show a more consistent memory usage.

### ISSUE THREAD

**Issue by krasi-georgiev** [prometheus/test-infra#324]
http://prombench.prometheus.io/grafana/d/7gmLoNDmz/prombench?orgId=1&var-pr-number=6729&from=1580931841961&to=1581143580452

With the Prom 2.16 release the rss and allocations show some difference and @bwplotka and @cstyan suggest that this is due to the way golang reports/uses/frees the memory with MADV_FREE

Bartek wrote a blog post about it with some suggestions to use different metrics  for a more realistic memory usage and we should try to use those to see if it will report a more accurate number.
https://www.bwplotka.dev/2019/golang-memory-monitoring/

**Comment by bwplotka** [comment]
:+1:

**Comment by bwplotka** [comment]
Also let's consider comments made on our IRC: https://matrix.to/#/!WaUKIfoqfiyWQhenET:matrix.org/$158134152150272vJGeo:matrix.org?via=matrix.org&via=half-shot.uk

**Comment by roidelapluie** [comment]
I would also appreciate the following dashboards dropdown variables:

- Being able to chose between rate and irate
- Being able to chose the rate inverval

**Comment by nevill** [comment]
Let me write down some ideas discussed on IRC.
1. Working Set Size / RSS doesn't really help, but we still care about it.
2. We need a solution to smooth the spiky lines, probably irate / rate, or max / avg / min ? It should be a deterministic way to tell how the memory is going.
3. As noted in the @bwplotka 's blog, there are two types of memory we need to care, `evictable` and `non-evictable`. We pay more attention to `non-evictable` in the case of promoting a release candidate.

I suggest that we use `go_memstats_heap_inuse_bytes` to represent the non-evictable and `go_memstats_heap_idle_bytes` to represent the evictable.
Basically, `inuse` = `memtstats_heap_alloc_bytes` + `fragmentation`(the memory allocated to runtime, but partially used). And `idle` = `memstats_heap_released_bytes` + `the amount can be returned to OS`.
BTW, we omit the stack size (also belongs to non-evictable) here, which is quite small compared with heap size.
